src/test/interfaces_tests.cpp

```diff
@@ -98,7 +98,7 @@ BOOST_AUTO_TEST_CASE(findAncestorByHash)
 BOOST_AUTO_TEST_CASE(findCommonAncestor)
 {
     auto& chain = m_node.chain;
-    const CChain& active = WITH_LOCK(Assert(m_node.chainman)->GetMutex(), return Assert(m_node.chainman)->ActiveChain());
+    const CChain& active{*WITH_LOCK(Assert(m_node.chainman)->GetMutex(), return &Assert(m_node.chainman)->ActiveChain())};
```

Comment:
I think it would be best to add the commit message as code documentation to make it easier to track when this can be reverted?

I'm also not sure this commit fits the scope of the PR but I can see how it's related, and it's a separate commit anyway so probably fine.

Reply:
GCC 13 introduced a broken behavior, and we usually do not modify our correct code to make a broken compiler happy. Considering that this is the only line in test code, I lean to agree with this change.

Reply:
As the test is execute in a single thread, is it big deal to hold `::cs_main` during the entire test case?

Why not to use the pattern as https://github.com/bitcoin/bitcoin/blob/fa266c4bbf564308ddbc12653527226506902084/src/test/interfaces_tests.cpp#L130-L132
?

Reply:
> Likely due to a LOCKS_EXCLUDED?

Right, in `Chainstate::InvalidateBlock`.